src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/RemoveAzureCdnCustomDomain.cs

```diff
@@ -0,0 +1,104 @@
+        [ValidateNotNullOrEmpty]
+        public string CustomDomainName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FieldsParameterSet, HelpMessage = "Azure Cdn endpoint name.")]
+        [ValidateNotNullOrEmpty]
+        public string EndpointName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FieldsParameterSet, HelpMessage = "Azure Cdn profile name.")]
+        [ValidateNotNullOrEmpty]
+        public string ProfileName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FieldsParameterSet, HelpMessage = "The resource group of the Azure Cdn Profile")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The profile.", ParameterSetName = ObjectParameterSet)]
+        [ValidateNotNull]
+        public PSCustomDomain CdnCustomDomain { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName == ObjectParameterSet)
+            {
+                EndpointName = CdnCustomDomain.EndpointName;
+                ProfileName = CdnCustomDomain.ProfileName;
+                ResourceGroupName = CdnCustomDomain.ResourceGroupName;
+                CustomDomainName = CdnCustomDomain.Name;
+            }
+
+            try
+            {
+                CdnManagementClient.CustomDomains.GetWithHttpMessagesAsync(
+                    CustomDomainName, 
+                    EndpointName, 
+                    ProfileName, 
+                    ResourceGroupName).Wait();
+            }
+            catch (AggregateException exception)
+            {
+                var errorResponseException = exception.InnerException as ErrorResponseException;
+                if (errorResponseException == null)
+                {
+                    throw;
+                }
+
+                if (errorResponseException.Response.StatusCode.Equals(HttpStatusCode.NotFound))
+                {
+                    throw new PSArgumentException(string.Format(Resources.Error_DeleteNonExistingCustomDomain,
+                        CustomDomainName,
+                        EndpointName,
+                        ProfileName,
+                        ResourceGroupName));
+                }
+            }
+
+            if (!ShouldProcess(string.Format(
+                            Resources.Confirm_RemoveCustomDomain,
+                            CustomDomainName,
+                            EndpointName,
+                            ProfileName,
+                            ResourceGroupName)))
+            {
+                return;
+            }
+
+            CdnManagementClient.CustomDomains.DeleteIfExists(
+                            CustomDomainName,
+                            EndpointName,
+                            ProfileName,
+                            ResourceGroupName);
+
+            WriteObject(true);
```

Comment:
Consider using a PassThrough parameter, so the cmdlet returns nothing if successful, or true if PassThrough is specified